src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1571,31 +1564,13 @@ void SolverInterfaceImpl::computeMappings(const utils::ptr_vector<MappingContext
 
 void SolverInterfaceImpl::mapData(DataContext &context, const std::string &mappingType)
 {
-  PRECICE_TRACE();
-  using namespace mapping;
-  MappingConfiguration::Timing timing;
-
-  if (not context.hasMapping()) {
-    return;
-  }
-
-  timing         = context.mappingContext().timing;
-  bool hasMapped = context.mappingContext().hasMappedData;
-  bool mapNow    = timing == MappingConfiguration::ON_ADVANCE;
-  mapNow |= timing == MappingConfiguration::INITIAL;
-
-  if ((not mapNow) || hasMapped) {
-    return;
-  }
-
-  int inDataID  = context.getFromDataID();
-  int outDataID = context.getToDataID();
+  int fromDataID = context.getFromDataID();
+  int toDataID   = context.getToDataID();
   PRECICE_DEBUG("Map \"{}\" data \"{}\" from mesh \"{}\"",
                 mappingType, context.getDataName(), context.getMeshName());
   context.resetToData();
-  PRECICE_DEBUG("Map from dataID {} to dataID: {}", inDataID, outDataID);
-  context.mappingContext().mapping->map(inDataID, outDataID);
-  PRECICE_DEBUG("Mapped values = {}", utils::previewRange(3, context.toData()->values())); // @todo might be better to move this debug message into Mapping::map and remove getter DataContext::toData()
```

Comment:
Why do you remove this DEBUG print? I think it might be useful. I think it might be useful.

Reply:
Good that you saw this. I removed it because it's the only place where `DataContext::toData()` is actually used. Maintaining this public function just for a debug statement is in my opinion too much and therefore I removed the function and the debug statement. Also fixed this now in 39910cc237a889e9a6cef862dd4e967e9f742bce and cleaned it up.

If somebody really needs to debug here, this can also happen somewhere else where this data is actually used and accessed, for example inside the mapping.

Reply:
IIRC we added these Debug statements here in a dedicated effort by @fsimonis 
These Debug statements are really important when developing an adapter. Let's please not remove them.

Reply:
Moved the debug statement inside `DataContext` then we also don't need `DataContext::toData()`.